car-bridge/src/main.rs

```diff
@@ -61,19 +62,24 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let drainage = Drainage::new();
 
     let (response_sender, mut response_receiver) = mpsc::channel(PUBLISH_BUFFER);
+    let mut response_sender = Some(response_sender);
 
     loop {
         select! {
-            message = messages.next() => {
+            message = messages.next(), if response_sender.is_some() => {
```

Comment:
nit: avoid `unwrap` by moving the guard inside:

```rust
message = messages.next() => {
    if let Some(ref response_sender) = response_sender {
        let Some(message) = message else {
            break;
        };
        spawn(handle_message(chariott.clone(), response_sender.clone(), message, cancellation_token.child_token()));
    }
}
```

I don't think there's anything to be gained by the guard on the `select!` arm since there isn't another `message = messages.next()` to disambiguate with.

Reply:
Addressed in f5629e7061003a686990331b34cbde8ca20c7b13